test/hotspot/jtreg/compiler/vectorization/TestPopulateIndex.java

```diff
@@ -81,14 +81,14 @@ public void checkResultIndexArrayFill() {
     @IR(counts = {IRNode.POPULATE_INDEX, "> 0"})
     public void exprWithIndex1() {
         for (int i = 0; i < count; i++) {
-            dst[i] = src[i] * (i & 7);
+            dst[i] = src[i] * (i | 7);
```

Comment:
This doesn't want to vectorize.

Reply:
As @merykitty said, the issue is probably that we split the `OrI` through the `phi`, and that breaks the pattern matching. Remove the IR rule for now, and add a comment like this:

`// Does not vectorize, possibly because the OrI is pushed through the Phi, see also JDK-8348096.`